lib/maths/CCalendarComponentAdaptiveBucketing.cc

```diff
@@ -276,7 +276,9 @@ void CCalendarComponentAdaptiveBucketing::refresh(const TFloatVec& oldEndpoints)
                 static_cast<double>(oldCentres[l - 1]));
             largeErrorCount += w * oldLargeErrorCounts[l - 1];
             count += w * w * CBasicStatistics::count(m_Values[l - 1]);
-            double scale{count / CBasicStatistics::count(value)};
+            double scale{count == CBasicStatistics::count(value)
```

Comment:
It's not so obvious here as in the other places that this is defending against `CBasicStatistics::count(value) == 0`, so it would be worth adding a comment like:

```
// Defend against division by zero - if CBasicStatistics::count(value) is zero then count must be too
```

(I am assuming that's the reasoning, but if I was reading this outside of a PR that was about avoiding division by zero it wouldn't immediately occur to me.)

Reply:
I added a comment in [da27f31](https://github.com/elastic/ml-cpp/pull/1225/commits/da27f31fbfea1d9590d7a083e4bfcb955aa17596).